hyperbahn/test/time-series/_time-series-cluster.js

```diff
@@ -0,0 +1,720 @@
+        ' should be greater than >= ' + options.min);
+    assert.ok(options.value <= options.max,
+        'in batch ( ' + options.index + ' ) count (' +
+        options.value + ') for' + options.description +
+        ' should be less than <= ' + options.max);
+};
+
+TimeSeriesCluster.prototype.takeHeaps =
+function takeHeaps(delayOne, delayTwo, delayThree) {
+    var HEAP_FILE_ONE = path.join(
+        __dirname,
+        path.basename(__filename).split('.')[0] + '-1first-heap.heapsnapshot'
+    );
+    var HEAP_FILE_TWO = path.join(
+        __dirname,
+        path.basename(__filename).split('.')[0] + '-2second-heap.heapsnapshot'
+    );
+    var HEAP_FILE_THREE = path.join(
+        __dirname,
+        path.basename(__filename).split('.')[0] + '-3third-heap.heapsnapshot'
+    );
+
+    var heapdump = require('heapdump');
+
+    setTimeout(firstHeap, delayOne * 1000);
+    setTimeout(secondHeap, delayTwo * 1000);
+    setTimeout(thirdHeap, delayThree * 1000);
+
+    function firstHeap() {
+        /* eslint no-console:0 */
+        console.log('writing first heap');
+        heapdump.writeSnapshot(HEAP_FILE_ONE);
+    }
+
+    function secondHeap() {
+        console.log('writing second heap');
+        heapdump.writeSnapshot(HEAP_FILE_TWO);
+    }
+
+    function thirdHeap() {
+        console.log('writing third heap');
+        heapdump.writeSnapshot(HEAP_FILE_THREE);
+    }
+};
+
+function BatchClient(options) {
+    if (!(this instanceof BatchClient)) {
+        return new BatchClient(options);
+    }
+
+    var self = this;
+    EventEmitter.call(self);
+
+    self.serverServiceName = options.serverServiceName;
+    self.clientServiceName = options.clientServiceName;
+    self.requestsPerBatch = options.requestsPerBatch;
+    self.clientBatchDelay = options.clientBatchDelay;
+    self.numBuckets = options.numBuckets;
+
+    self.numBatchesInBucket = options.numBatchesInBucket;
+
+    self.clientTimeout = options.clientTimeout;
+    self.endpoint = options.endpoint;
+
+    self.timeWindow = options.timeWindow;
+    self.channels = [];
+    for (var i = 0; i < options.remotes.length; i++) {
+        self.channels.push(options.remotes[i].clientChannel);
+    }
+
+    self.requestVolume = self.numBuckets * (
+        self.numBatchesInBucket * self.requestsPerBatch
+    );
+
+    self.body = options.body || '';
+    self.bodyChunks = options.bodyChunks || [];
+    self.requestOptions = {
+        serviceName: self.serverServiceName,
+        timeout: self.clientTimeout,
+        headers: {
+            'as': 'raw',
+            'cn': self.clientServiceName
+        }
+    };
+    self.streamingRequestOptions = {
+        serviceName: self.serverServiceName,
+        streamed: true,
+        timeout: self.clientTimeout,
+        headers: {
+            'as': 'raw',
+            'cn': self.clientServiceName
+        }
+    };
+    self.streamWriteDelay = 10;
+
+    self.freeList = new Array(self.requestVolume);
```

Comment:
I don’t believe there’s a win to constructing an array with an initial size. You don’t have to do this to avoid push function calls.